front: simplify formatSimulationTable

clarani opened the pull request:
Handle first the stdcm case and then the operational studies case. I've decoupled the logic to remove a lot of tests/conditions, and added a function to generate the row style.

clarani committed:
front: simplify formatSimulationTable

Handle first the stdcm case and then the operational studies case.
I've decoupled the logic to remove a lot of tests/conditions, and added
a function to generate the row style.

Signed-off-by: Clara Ni <clara.ni@outlook.fr>

clarani committed:
front: split the formatSimulationTable function

This commit is only about moving code around. This split will be useful
for the following commits.

Signed-off-by: Clara Ni <clara.ni@outlook.fr>

clarani committed:
front: simplify formatSimulationTable props

Signed-off-by: Clara Ni <clara.ni@outlook.fr>

clarani committed:
front: harmonize SimulationTable props

SimulationTable takes now the same props in both modes (stdcm or operationalStudies).
The table data is formatted outside the SimulationTable component.

Signed-off-by: Clara Ni <clara.ni@outlook.fr>

SharglutDev commented on front/src/modules/SimulationReportSheet/utils/formatSimulationTable.ts:
```diff
@@ -56,95 +63,77 @@ const getRowStyle = (
-    others: Style;
-  };
-};
+const formatStdcmDataForSimulationTable = (
```
nit: if the function uses React (here `useTranslation)`, its name should start with `use...` otherwise we should pass `t` as a param (`TFunction<'stdcm'>`). Same for the other function.

clarani replied:
Done !

SharglutDev reviewed (commented):
Just passing by, didn't review everything

clarani committed:
front: fix simulation report sheet translation

Signed-off-by: Clara Ni <clara.ni@outlook.fr>

clarani committed:
front: make duration nullable in OperationalPointWithTimeAndSpeed

Now that duration can be nullable, we can make the distinction between a stop of 0s and no stop.

Signed-off-by: Clara Ni <clara.ni@outlook.fr>

theocrsb reviewed (approved):
LGTM and tested !

kmer2016 reviewed (approved):
LGTM ! Very nice ;)
